File: problem2.rb
# sum of even numbers in fibonacci sequence with value < 4,000,000 (n)

def problem2(n)
  fibonacciNumbers = [1, 2]
  while fibonacciNumbers.max < n
    nextNum = fibonacciNumbers[-1] + fibonacciNumbers[-2]
    fibonacciNumbers.push(nextNum)
  end
  fibonacciNumbers.pop
  return sumIfEven(fibonacciNumbers)
end

def sumIfEven(arr)
  arr.select{ |n| n.even? }.sum
end

File: problem2.js
// sum of even numbers in fibonacci sequence with value < 4,000,000 (n)

const problem2 = n => {
  let numToSum = [1, 2];
  while (Math.max(...numToSum) < n) {
    nextNumber = numToSum.slice(-2).reduce(reducer);
    numToSum.push(nextNumber);
  }
};

const reducer = (accumulator, currentValue) => accumulator + currentValue;

const isEven = x => {
  return x % 2 === 0 ? true : false;
};
